### DISCUSSION TITLE
classes/class_audiosample

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#145] [User-contributed notes]
# classes/class_audiosample

Experimental: This class may be changed or removed in future versions. Inherits: RefCounted< Object Base class for audio samples. Description: Base class for audio samples.

https://docs.godotengine.org/en/stable/classes/class_audiosample.html

**Comment by jonathandavidlewis** [comment]
Note that `AudioSample` playback type does not support `AudioEffect`.
`AudioSample` playback type is also now the default type for web export in 4.3.
This means that by default, `AudioEffect`s applied to audio channels will not work on web builds.

There are some options including changing the default to `Stream` in the project settings and using multithreaded web build, but determine what is best for you.

**Reply by jonathandavidlewis** [reply]
This is mentioned in the tooltop for an `AudioStreamPlayer` `playback_type` field. You can see the source of the tooltip here:

https://github.com/godotengine/godot-docs/blob/3a0758017759d528c22dd91f15417900d69f75eb/classes/class_audioserver.rst#id279
